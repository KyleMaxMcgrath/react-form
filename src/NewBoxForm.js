import React, {useState} from 'react';

const NewBoxForm = ({submit}) => {
    const INITIAL_DATA = {
        width: 100,
        height: 100,
        backgroundColor: '#0000FF'
    };
    const [formData, setFormData] = useState(INITIAL_DATA);

    const handleChange = (e) => {
        const {name, value} = e.target
        setFormData(
            {
                ...formData,
                [name]: value
            }
        )
    }

    const onSubmit = (e) => {
        e.preventDefault();
        submit(formData);
    }

    return (
        <form onSubmit={onSubmit}>
            <div className="NewBoxForm-inputGroup">
                <label htmlFor="width">Width: {formData.width}</label>
                <input name="width" value={formData.width} type="range" onChange={handleChange} max={500} min={1}/>
            </div>
            <br/>
            <div className="NewBoxForm-inputGroup">
                <label htmlFor="height">Height: {formData.height}</label>
                <input name="height" value={formData.height} type="range" onChange={handleChange} max={500} min={1}/>
            </div>
            <br/>
            <div className="NewBoxForm-inputGroup">
                <label htmlFor="backgroundColor"> Background Color: </label>
                <input name="backgroundColor" value={formData.backgroundColor} type="color" onChange={handleChange} max={500} min={1}/>
            </div>
            <br/>
            <button>Add Box</button>
        </form>
    )

}

export default NewBoxForm;