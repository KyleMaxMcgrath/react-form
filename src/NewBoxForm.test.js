import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('renders', () => {
  render(<NewBoxForm />);
});

it("matches snapshot", function() {
    const {asFragment} = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});