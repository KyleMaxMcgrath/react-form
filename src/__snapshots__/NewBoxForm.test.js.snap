// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="NewBoxForm-inputGroup"
    >
      <label
        for="width"
      >
        Width: 100
      </label>
      <input
        max="500"
        min="1"
        name="width"
        type="range"
        value="100"
      />
    </div>
    <br />
    <div
      class="NewBoxForm-inputGroup"
    >
      <label
        for="height"
      >
        Height: 100
      </label>
      <input
        max="500"
        min="1"
        name="height"
        type="range"
        value="100"
      />
    </div>
    <br />
    <div
      class="NewBoxForm-inputGroup"
    >
      <label
        for="backgroundColor"
      >
         Background Color: 
      </label>
      <input
        max="500"
        min="1"
        name="backgroundColor"
        type="color"
        value="#0000FF"
      />
    </div>
    <br />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
